src/ports/postgres/modules/mxgboost/madlib_xgboost.py_in

```diff
@@ -0,0 +1,549 @@
+        else:
+            r = ["""{k}={v}""".format(k=key,v=val)]
+        params_list.append(r)
+    params_grid = [l for l in itertools.product(*params_list)]
+    return params_grid
+
+def xgboost_train(schema_madlib, dframe, features_all, class_label, params,
+                  class_weights, train_set_size, id_column, train_set_split_var):
+
+    df = pickle.loads(zlib.decompress(dframe))
+    features_all.append(id_column)
+    features = filter(lambda x: x in df.columns, features_all)
+    X = df[features].as_matrix()
+    y = df[class_label]
+    class_list = numpy.unique(y).tolist()
+
+    if not train_set_split_var or train_set_split_var == 'None':
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=float(1-train_set_size))
+        #We don't actually want the test set size to change. We want it to be constant as we change train set size so we can compare apples to apples
+        #so lets lock it at 20% (only less if the train size is > 80%)
+        test_set_size = min((1-train_set_size),0.2)
+        X_test = X_test[range(0,int(len(y)*test_set_size)),]
+        y_test = y_test.head(int(len(y)*test_set_size))
+    else:
+        split_indx = numpy.WHERE(features == train_set_split_var)[0]
+        X = numpy.delete(X,split_indx,1)
+        X_train = X[numpy.array(df[train_set_split_var]==1),]
+        X_test = X[numpy.array(df[train_set_split_var]==0),]
+        y_train = y[numpy.array(df[train_set_split_var]==1)]
+        y_test = y[numpy.array(df[train_set_split_var]==0)]
+    #save off and remove the id_column for later output. Make sure to get rid of id_column from features!
+    test_ids = X_test [:,len(features)-1]
+    X_train = numpy.delete(X_train,len(features)-1,1)
+    X_test = numpy.delete(X_test,len(features)-1,1)
+    features = features[0:len(features)-1]
+
+    class_list_y_train = numpy.unique(y_train).tolist()
+    class_list_y_test = numpy.unique(y_test).tolist()
+    if (class_list != class_list_y_train) or (class_list != class_list_y_test):
+        plpy.error("Train test split caused a subset with missing classes.")
+
+    #run weights
+    sample_representation = y_train.value_counts()
+    total_samples = sum(sample_representation)
+    sample_weights = None
+    if not class_weights:
+        sample_weights = map(
+                lambda s: total_samples*1.0/sample_representation[s]
+                                /
+                sum([total_samples*1.0/sample_representation[c] for c in sample_representation.keys()])
+                ,
+                y_train
+            )
+    else:
+        #User-supplied class-weights
+        class_weights_dict = ast.literal_eval(re.sub("[\\t]","",class_weights).strip())
+        sample_weights = map(lambda s: class_weights_dict[s], y_train)
+
+    #Train gradient boosted trees
+    p_list = [p.split('=') for p in ast.literal_eval(re.sub("[\\t]","",params).strip())]
+    params_dict = dict([(k, ast.literal_eval(v.strip())) for k,v in p_list])
```

Comment:
we do not validate any values passed in params_dict. 
```
$$ 
{ 
'learning_raet': [0.01, 0.1]#Regularization on weights (eta). For smaller values, increase n_estimators 
'max_depth': [9,12],#Larger values could lead to overfitting
'subsampl': [0.85],#introduce randomness in samples picked to prevent overfitting
'colsample_bytree': [0.85],#introduce randomness in features picked to prevent overfitting
'min_child_weight': [10],#larger values will prevent over-fitting
'n_estimators':[100] #More estimators, lesser variance (better fit on test set)
}                                                                                                                                                            
$$
``` 

what if there is a typo, instead of learning_rate => learning_raet. Currently it doesn't fail in the `XGBClassifier()` call, so don't know what it's interpreted as. 
Not sure if  we want to handle this, but this may probably be just something that goes unnoticed.

Reply:
I tested xgboost directly and it does not fail if there is an unexpected parameter. I think it is OK to preserve that behavior and mention it in the documentation.

Reply:
Agreed! I think its a good idea to document it.